### PR TITLE
Selection improvement after inserting a horizontal rule

### LINKED ISSUE #3: Selection should be moved to the next line after inserting a horizontal rule
Two reasons:

* It doesn't make sense to select the HR. There's simply no use for that.
* There's a problem when you insert a HR at the end of the document (see https://github.com/ckeditor/ckeditor5/issues/407) that you may not know how to insert a paragraph after it. You can actually do that by pressing <kbd>Enter</kbd> but if we insert that paragraph automatically this will be better.

### LINKED ISSUE #5516: Selection should be moved to the next line after inserting a horizontal rule
Two reasons:

* It doesn't make sense to select the HR. There's simply no use for that.
* There's a problem when you insert a HR at the end of the document (see https://github.com/ckeditor/ckeditor5/issues/407) that you may not know how to insert a paragraph after it. You can actually do that by pressing <kbd>Enter</kbd> but if we insert that paragraph automatically this will be better.

### PR DESCRIPTION
**Pull request by pomek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Internal: Improved a mechanism that handles the selection after inserting a horizontal rule. Closes ckeditor/ckeditor5#5516.

### CODE REVIEW

File: src/horizontalrulecommand.js [hunk 1]
```diff
@@ -38,9 +38,32 @@ export default class HorizontalRuleCommand extends Command {
...
+			// If the element is missing or it's not a paragraph, but the paragraph could be inserted
+			// next to the horizontal rule, let's add it.
+			if ( !( nextElement && nextElement.is( 'paragraph' ) ) && model.schema.checkChild( horizontalElement.parent, 'paragraph' ) ) {
+				nextElement = writer.createElement( 'paragraph' );
+
+				writer.insert( nextElement, writer.createPositionAfter( horizontalElement ) );
+			}
+
+			// Put the selection at the beginning of the element.
+			if ( nextElement ) {
```
Comment (jodator) [inline src/horizontalrulecommand.js:58]: It seems that we do the same check twice but differently. I've made passing test with this:

```js
const canSetSelection = nextElement && model.schema.checkChild( nextElement, '$text' );

if ( !canSetSelection && model.schema.checkChild( horizontalElement.parent, 'paragraph' ) ) {
	nextElement = writer.createElement( 'paragraph' );

	writer.insert( nextElement, writer.createPositionAfter( horizontalElement ) );
}

if ( nextElement ) {
	writer.setSelection( nextElement, 0 );
}
```

**Review by jodator — CHANGES_REQUESTED** [review]
So I can see only one thing in the code which you might fix (see comment).

~The second thing is what should happen on delete? Probably it is a separate ticket since I couldn't find anything in this regard.~ Nah it's cool- other widgets behave the same: an empty paragraph is created. If I find this annoying later I'll report this.

ps.: Check the missing dev deps:

```
Missing devDependencies:

- @ckeditor/ckeditor5-engine
- @ckeditor/ckeditor5-utils
```

**Comment by pomek** [comment]
@jodator, fixed.

**Review by jodator — APPROVED** [review]
LGTM now.

**Pull request merged**